### DISCUSSION TITLE
Import `DeepPartial` from `.ts` instead declaring via `.d.ts`

### DISCUSSION THREAD

**Discussion by andrenarchy** [directus/directus#5838] [Feature Requests]
Declaring `DeepPartial` (and potentially other types) in a `.d.ts` file works for developing inside the directus project. However, the Typescript compiler doesn't know about it if Directus is imported in a downstream package. I suggest renaming the file to a `.ts` file and to import `DeepPartial` where required.

**Comment by rijkvanzanten** [comment]
I think the answer to both this and your other idea (https://github.com/directus/directus/discussions/5837) could be to have a shared package that's explicitly used to share the types and utils you'd use in extensions (https://github.com/directus/directus/discussions/5837) 🙂

**Reply by andrenarchy** [reply]
That'd be a good solution indeed! :) I'm currently importing several files directly like this:

```
import * as services from 'directus/dist/services'
import storage from 'directus/dist/storage'
import { getSchema } from 'directus/dist/utils/get-schema'
```

By the way, the `directus` npm package doesn't seem to export anything at the moment... maybe the extension types could go there so we can do `import {...} from 'directus'`?

**Comment by nickrum** [comment]
`DeepPartial` has since been moved to a `.ts` file.
